src/cmap/worker.rs

```diff
@@ -173,7 +205,7 @@ impl ConnectionPoolWorker {
 
         loop {
             let task = tokio::select! {
-                Some(request) = self.request_receiver.recv(), if self.can_service_connection_request() => {
```

Comment:
Because we need to be able to clear the WaitQueue now, we need to buffer the requests into a separate data structure instead of leaving them in the channel.